js/viz/axes/base_axis.js

```diff
@@ -2187,8 +2188,9 @@ Axis.prototype = {
                 maxVisible: previousRange.endValue,
                 categories: previousRange.categories
             };
-            const shift = adjust(that.getVisualRangeCenter() - that.getVisualRangeCenter(previousBusinessRange));
-            const zoomFactor = +(Math.round(that.getVisualRangeLength(previousBusinessRange) / that.getVisualRangeLength() + "e+2") + "e-2");
+            const typeIsChanged = that.getOptions().type !== that._storedZoomEndParams.type;
```

Comment:
Maybe invert this comparison? It can follow to direct condition [here](https://github.com/DevExpress/DevExtreme/pull/9076/files#diff-29be1e1b2908f00b9627c1252c7adfa4R2192)